internal/provisioner/controller/gatewayclass.go

```diff
@@ -227,70 +239,103 @@ func (r *gatewayClassReconciler) Reconcile(ctx context.Context, req ctrl.Request
-	status metav1.ConditionStatus,
-	reason gatewayapi_v1beta1.GatewayClassConditionReason,
-	message string,
-) error {
-	currentAcceptedCondition := metav1.Condition{
-		Type:               string(gatewayapi_v1.GatewayClassConditionStatusAccepted),
-		Status:             status,
-		ObservedGeneration: gatewayClass.Generation,
-		LastTransitionTime: metav1.Now(),
-		Reason:             string(reason),
-		Message:            message,
-	}
-	var newConds []metav1.Condition
-	for _, cond := range gatewayClass.Status.Conditions {
-		if cond.Type == string(gatewayapi_v1.GatewayClassConditionStatusAccepted) {
-			if cond.Status == status {
+func (r *gatewayClassReconciler) setConditions(ctx context.Context, gatewayClass *gatewayapi_v1beta1.GatewayClass, newConds map[string]metav1.Condition) error {
+	var unchangedConds, updatedConds []metav1.Condition
+	for _, existing := range gatewayClass.Status.Conditions {
+		if cond, ok := newConds[existing.Type]; ok {
+			if existing.Status == cond.Status {
 				// If status hasn't changed, don't change the condition, just
 				// update the generation.
-				currentAcceptedCondition = cond
-				currentAcceptedCondition.ObservedGeneration = gatewayClass.Generation
+				changed := existing
+				changed.ObservedGeneration = gatewayClass.Generation
+				updatedConds = append(updatedConds, changed)
+				delete(newConds, cond.Type)
 			}
-
-			continue
+		} else {
+			unchangedConds = append(unchangedConds, existing)
 		}
-
-		newConds = append(newConds, cond)
 	}
 
-	r.log.WithValues("gatewayclass-name", gatewayClass.Name).Info(fmt.Sprintf("setting gateway class's Accepted condition to %s", status))
+	transitionTime := metav1.Now()
+	for _, c := range newConds {
+		r.log.WithValues("gatewayclass-name", gatewayClass.Name).Info(fmt.Sprintf("setting gateway class's %s condition to %s", c.Type, c.Status))
+		c.ObservedGeneration = gatewayClass.Generation
+		c.LastTransitionTime = transitionTime
+		updatedConds = append(updatedConds, c)
+	}
 
 	// nolint:gocritic
-	gatewayClass.Status.Conditions = append(newConds, currentAcceptedCondition)
+	gatewayClass.Status.Conditions = append(unchangedConds, updatedConds...)
 
 	if err := r.client.Status().Update(ctx, gatewayClass); err != nil {
-		return fmt.Errorf("failed to set gatewayclass %s accepted condition: %w", gatewayClass.Name, err)
+		return fmt.Errorf("failed to update gateway class %s status: %w", gatewayClass.Name, err)
 	}
-
 	return nil
 }
 
+func (r *gatewayClassReconciler) getSupportedVersionCondition(ctx context.Context) metav1.Condition {
+	cond := metav1.Condition{
+		Type: string(gatewayapi_v1.GatewayClassConditionStatusSupportedVersion),
+		// Assume false until we get to the happy case.
+		Status: metav1.ConditionFalse,
+		Reason: string(gatewayapi_v1.GatewayClassReasonUnsupportedVersion),
+	}
+	gatewayClassCRD := &apiextensions_v1.CustomResourceDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "gatewayclasses." + gatewayapi_v1.GroupName,
```

Comment:
right now this is just looking at a single resource (gatewayclasses), we could look at them all if present, but (in the future) that would also mean we have to handle the case where a user might request to only reconcile standard CRDs so we wouldn't want to check those

Reply:
I'm comfortable with this as-is for now